From f873aa3ec533e20d5c70e9a6151f575472f50f75 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Sun, 07 Aug 2022 10:37:25 +0300
Subject: [PATCH] [SW-96509] preboot: hl: fw_info: clear device activity counter

The device activity counter, which indicates the number
of times the device was used by the user from the last
cold reset or prstn reset, this data is provided to fw
from LKD and reported in RAS, this commit clears the counter
whenever cold reset or perstn occur.

Change-Id: Ic294e341bdfb9400c33338650e6f1e53bf00f6ba
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index 674a2f1..9d563c9 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -404,6 +404,10 @@
 		boot_info_get(HL_BOOT_INFO_FW_UPD_CLOSURE, &fw_upd_is_dis,
 			      sizeof(fw_upd_is_dis), &data_size);
 
+	/* device activity counter is cleared in power cycle or prstn */
+	if (rst_src == HL_PRSTN_RST || rst_src == HL_COLD_RST)
+		fw_info.device_activity_counter = 0;
+
 	module_fwupd_pcie_disable(fw_upd_is_dis);
 
 	if (!boot_info_get(HL_BOOT_INFO_FW_INFO, &fw_info,
